bt_editor/mainwindow.h

```diff
@@ -42,7 +42,11 @@ class MainWindow : public QMainWindow
                               SUBTREE_REFRESH};
 
 public:
-    explicit MainWindow(GraphicMode initial_mode, QWidget *parent = nullptr);
+    explicit MainWindow(GraphicMode initial_mode,
+                       const QString& monitor_pub_port = "",
+                       const QString& monitor_srv_port = "",
```

Comment:
Could also use `std::optional` here if preferred (dunno if we're cool with C++17)